Stratis.Bitcoin.IntegrationTests/APITests.cs

```diff
@@ -65,50 +66,59 @@ public void CanGetGeneralInfoViaAPI()
-                    var mnemonic = walletManager.CreateWallet(model.Password, model.Name);
-
-                    var content = new StringContent(model.ToString(), Encoding.UTF8, "application/json");
-                    var response = client.PostAsync(ApiURI + "api/miner/startstaking", content).GetAwaiter().GetResult();
-                    Assert.Equal(System.Net.HttpStatusCode.OK, response.StatusCode);
-
-                    var responseText = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
-                    Assert.Equal("", responseText);
-
-                    MiningRPCController controller = fullNode.NodeService<MiningRPCController>();
-                    GetStakingInfoModel info = controller.GetStakingInfo();
-
-                    Assert.NotNull(info);
-                    Assert.Equal(true, info.Enabled);
-                    Assert.Equal(false, info.Staking);
+                    CoreNode nodeA = builder.CreateStratisPosNode(false, fullNodeBuilder =>
+                    {
+                        fullNodeBuilder
+                       .UseStratisConsensus()
+                       .UseBlockStore()
+                       .UseMempool()
+                       .UseWallet()
+                       .AddPowPosMining()
+                       .UseApi()
+                       .AddRPC();
+                    });
+
+                    builder.StartAll();
+
+                    var fullNode = nodeA.FullNode;
+                    var ApiURI = fullNode.Settings.ApiUri;
+
+                    Assert.NotNull(fullNode.NodeService<PosMinting>(true));
+
+                    using (Node nodeB = nodeA.CreateNodeClient())
+                    {
+                        WalletManager walletManager = fullNode.NodeService<IWalletManager>() as WalletManager;
+
+                        // create the wallet
+                        var model = new StartStakingRequest() { Name = "apitest", Password = "123456" };
+                        var mnemonic = walletManager.CreateWallet(model.Password, model.Name);
+
+                        var content = new StringContent(model.ToString(), Encoding.UTF8, "application/json");
+                        var response = client.PostAsync(ApiURI + "api/miner/startstaking", content).GetAwaiter().GetResult();
+                        Assert.Equal(System.Net.HttpStatusCode.OK, response.StatusCode);
+
+                        var responseText = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                        Assert.Equal("", responseText);
+
+                        MiningRPCController controller = fullNode.NodeService<MiningRPCController>();
+                        GetStakingInfoModel info = controller.GetStakingInfo();
+
+                        Assert.NotNull(info);
+                        Assert.Equal(true, info.Enabled);
+                        Assert.Equal(false, info.Staking);
+                    }
                 }
             }
+            finally
```

Comment:
Have a look at the WalletManagerTest file, we revert the side-effects by calling the Dispose() method.
This method will be called begore every test, so it has the advantage of being defined once for all the tests in the class.

Like so
```
public void Dispose()
{
    Transaction.TimeStamp = false;
    Block.BlockSignature = false;
}
```

Reply:
It's important that this happen explicitly instead of waiting for the GC to call Dispose. I have combined the two approaches by calling Dispose in finally and moving my code into Dispose.

Reply:
did you do it? I can't see any commit.

Reply:
I've added it to the next PR to avoid a merge clash and it is required there too.

Reply:
ok cool, thanks.